grunt/service-worker-fetch-listener.js

```diff
@@ -0,0 +1,57 @@
+self.addEventListener('fetch', function(event) {
+  if (event.request.method === 'GET') {
+    if (event.request.headers.get('accept').indexOf('application/json') !== -1) {
+      event.respondWith(async function() {
+        return fetch(event.request);
+      }());
```

Comment:
Can we just `return;` here? eg:

```js
if (event.request.headers.get('accept').indexOf('application/json') !== -1) {
  return;
}
```

I think the default behaviour is to do the fetch, so if we do nothing it'll work as expected.

Reply:
Yes, the default behaviour is to do fetch, actually prior to that I just made a simple expression i.e.

```js
if (event.request.method === 'GET' &&
      event.request.headers.get('accept').indexOf('application/json') === -1) {
```

This works perfectly but for some strange reasons e2e test ends up doing a fetch that returns the cached html page instead of the making a request and giving us json response.

It is quite a shame, I now have to bend the code like this to make the test happy. But nonetheless I will add a comment to that piece of code to create awareness of why we are doing that

I first encountered these disparity between real world and test environment when I was levering on service worker's  _runtimeCaching_ and _ignoreUrlParametersMatching [Array⟨Regex⟩]_

Reply:
The comment helps, thanks!